src/ifcopenshell-python/docs/ifcopenshell/getting_started.rst

```diff
@@ -0,0 +1,109 @@
+        for (typename Schema::IfcProduct::list::it it = elements->begin();
+            it != elements->end(); ++it) {
+            Schema::IfcProduct *ifcProduct = *it;
+            // TODO: Do something with ifcProduct
+        }
+    }
+    //
+    int main(int argc, char* argv[]) {
+        // file of IfcParse::IfcFile previously defined
+        auto schema_version = file.schema()->name();
+        schema_version = schema_version.substr(3);
+        std::transform(schema_version.begin(), schema_version.end(), schema_version.begin(), [](unsigned char c) { return std::tolower(c); });
+        if (schema_version == "4x3_rc2") {
+            parseIfc<Ifc4x3_rc2>(file);
+        }
+        else if (schema_version == "4") {
+            parseIfc<Ifc4>(file);
+        }
+        else if (schema_version == "2x3") {
+            parseIfc<Ifc2x3>(file);
+        }
+        // TODO: Handle more cases of IFC schema versions here
+    }
+
+Reading out attributes of an IfcProduct
+---------------------------------------
+
+The properties of an ``IfcProduct``, and by extension any derived class, can be read as by calling the method of the same name, e.g. ``GlobalId()``, ``Name()``. 
+Note that optional properties like name, long name, or description, among others, are wrapped with a ``boost::optional`` (`documentation <https://www.boost.org/doc/libs/1_84_0/libs/optional/doc/html/index.html>`__) container object.
+
+Navigating relationships in IFC
+-------------------------------
+
+While it is natural to look at the IFC format as a representation of a physical building model where physical objects are related to each other in a tree structure (e.g. site > building > storey > slab),
+the IFC format allows for the representation of information in a graph-like manner, joining physical objects to meta-information through relationships. 
+A detailed explanation of relationships in IFC is provided in this `blog post <https://constructingdata.wordpress.com/2018/04/09/ifc-for-the-layman-part-3-relationships/>`__. 
+
+The following function shows how property sets can be extracted from a given ``IfcObject``. 
+
+.. code-block:: c++
+
+    template<typename Schema>
+    void extractPropertySets(const typename Schema::IfcObject &obj,
+                     typename std::vector<typename Schema::IfcPropertySet *> &property_sets) {
+        const auto &is_defined_by = obj.IsDefinedBy();
+        if (is_defined_by == nullptr)
+            return;
+        std::vector<typename Schema::IfcRelDefines *> rels(is_defined_by->size());
+        typename std::vector<typename Schema::IfcRelDefines *>::iterator it = std::copy_if(
+            is_defined_by->begin(), is_defined_by->end(), rels.begin(),
+            [](const typename Schema::IfcRelDefines *x) {
+                if (x == nullptr) return false;
+                const typename Schema::IfcRelDefinesByProperties *defines_by_properties =
```

Comment:
I would use `x->template as<typename Schema::IfcRelDefinesByProperties>()` here.

Not sure if `template as` is required or `as` is sufficient. We need to check with gcc/clang.

Reply:
Same for below. Since we're setting the basis here, better do as<...> instead of dynamic_cast<...>. Functionally it's the same, but we might implement it differently later on.

Reply:
`template as` is necessary. `as` doesn't seem to compile